src/main/kotlin/com/openlattice/shuttle/Shuttle.kt

```diff
@@ -95,15 +115,58 @@ class Shuttle(
         }
     }
 
-    private val uploadRegulator = Semaphore(threads)
+    //vars to be used only when shuttle is run on shuttle server
+    private var writeLog: (String, String, OffsetDateTime, IntegrationStatus) -> Unit
+    private lateinit var logEntitySet: EntitySet
+    private lateinit var logsDestination: PostgresDestination
+    private lateinit var integrationJobs: IMap<UUID, IntegrationJob>
+    private val logProperties = mutableMapOf<FullQualifiedName, PropertyType>()
+    private val ptidsByBlackboxProperty = mutableMapOf<BlackboxProperty, UUID>()
+
+    init {
+        if (blackbox.enabled) {
+            val jobId = maybeJobId.get()
+            integrationJobs = hazelcastInstance!!.getMap(HazelcastMap.INTEGRATION_JOBS.name)
+
+            this.writeLog = { name, log, time, status ->
+                logger.info(log)
+                storeLog(name, log, time, status, jobId)
+                integrationJobs.executeOnKey(jobId, UpdateIntegrationStatusEntryProcessor(status))
+            }
+
+            blackbox.fqns.forEach {
+                val fqn = FullQualifiedName(it.value)
+                val propertyType = propertyTypes.getValue(fqn)
+                logProperties[fqn] = propertyType
+                ptidsByBlackboxProperty[it.key] = propertyType.id
+            }
+
+            logEntitySet = maybeLogEntitySet.get()
+            val logEntityTypeId = logEntitySet.entityTypeId
+            val logDataSource = HikariDataSource(HikariConfig(parameters.postgres.config))
+            logDataSource.maximumPoolSize = 5
```

Comment:
should be extracted to/supplied by configuration if possible. To a constant in this class if not

Reply:
:upvote:

Reply:
➕